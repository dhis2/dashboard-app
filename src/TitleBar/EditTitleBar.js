import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { t } from 'i18next';

import ItemSelect from '../ItemSelect/ItemSelect';
import D2ContentEditable from '../widgets/D2ContentEditable';
import { fromEditDashboard } from '../actions';

const EditTitleBar = ({
    name,
    description,
    style,
    onChangeTitle,
    onChangeDescription,
}) => {
    const titleBarEdit = Object.assign({}, style.titleBar, {
        justifyContent: 'space-between',
    });

    return (
        <Fragment>
            <span>Currently editing</span>
            <div style={titleBarEdit}>
                <div style={style.title}>
                    <D2ContentEditable
                        className="dashboard-title editable-text"
                        text={name}
                        disabled={false}
                        placeholder={t('Add title here')}
                        onChange={onChangeTitle}
                    />
                </div>
                <ItemSelect />
            </div>
            <div style={style.description}>
                <D2ContentEditable
                    className="dashboard-description editable-text"
                    text={description}
                    disabled={false}
                    placeholder={t('Add description here')}
                    onChange={onChangeDescription}
                />
            </div>
        </Fragment>
    );
};

const mapDispatchToProps = dispatch => ({
    onChangeTitle: event =>
        dispatch(fromEditDashboard.acSetDashboardTitle(event.target.value)),
    onChangeDescription: event =>
        dispatch(
            fromEditDashboard.acSetDashboardDescription(event.target.value)
        ),
});

const TitleBarCt = connect(null, mapDispatchToProps)(EditTitleBar);

export default TitleBarCt;

EditTitleBar.propTypes = {
    name: PropTypes.string,
    description: PropTypes.string,
};

EditTitleBar.defaultProps = {
    name: '',
    description: '',
};
