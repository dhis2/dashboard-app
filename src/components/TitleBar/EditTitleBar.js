import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import i18n from '@dhis2/d2-i18n'
import { InputField, TextAreaField } from '@dhis2/ui-core'

import ItemSelector from '../ItemSelector/ItemSelector'
import {
    acSetDashboardTitle,
    acSetDashboardDescription,
} from '../../actions/editDashboard'
import { orObject } from '../../modules/util'
import { sGetEditDashboardRoot } from '../../reducers/editDashboard'

import classes from './styles/EditTitleBar.module.css'

export const EditTitleBar = ({
    name,
    description,
    onChangeTitle,
    onChangeDescription,
}) => {
    const updateTitle = (_, e) => {
        onChangeTitle(e.target.value)
    }

    const updateDescription = (_, e) => {
        onChangeDescription(e.target.value)
    }

    return (
        <section className={classes.section}>
            <div className={classes.titleDescription}>
                <InputField
                    className={classes.title}
                    name="Dashboard title input"
                    label={i18n.t('Dashboard title')}
                    type="text"
                    onChange={updateTitle}
                    value={name}
                />
                <TextAreaField
                    className={classes.description}
                    name="Dashboard description input"
                    label={i18n.t('Dashboard description')}
                    onChange={updateDescription}
                    value={description}
                />
            </div>
            <div className={classes.itemSelector}>
                <ItemSelector />
            </div>
        </section>
    )
}

EditTitleBar.propTypes = {
    onChangeDescription: PropTypes.func.isRequired,
    onChangeTitle: PropTypes.func.isRequired,
    description: PropTypes.string,
    name: PropTypes.string,
}

EditTitleBar.defaultProps = {
    name: '',
    description: '',
}

const mapStateToProps = state => {
    const selectedDashboard = orObject(sGetEditDashboardRoot(state))

    return {
        name: selectedDashboard.name,
        description: selectedDashboard.description,
    }
}

const mapDispatchToProps = {
    onChangeTitle: acSetDashboardTitle,
    onChangeDescription: acSetDashboardDescription,
}

export default connect(mapStateToProps, mapDispatchToProps)(EditTitleBar)
