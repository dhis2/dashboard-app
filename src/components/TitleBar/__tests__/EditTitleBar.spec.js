import React from 'react';
import { shallow } from 'enzyme';
import { EditTitleBar } from '../EditTitleBar';

jest.mock('@dhis2/d2-ui-core/text-field/TextField', () => 'textfield');
jest.mock('../../ItemSelector/ItemSelector', () => 'itemselector');

describe('EditTitleBar', () => {
    const props = {
        name: 'Rainbow Dash',
        description: 'The blue one',
        onChangeTitle: Function.prototype,
        onChangeDescription: Function.prototype,
        classes: {
            section: 'section',
            titleDescription: 'titledesc',
            title: 'title',
            description: 'description',
            underline: 'underline',
            input: 'input',
            itemSelector: 'itemSelector',
        },
    };

    it('renders correctly when displayName not provided', () => {
        const tree = shallow(<EditTitleBar {...props} />);
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly when displayName is provided', () => {
        const tree = shallow(
            <EditTitleBar displayName="Regnbue Dash" {...props} />
        );
        expect(tree).toMatchSnapshot();
    });
});
