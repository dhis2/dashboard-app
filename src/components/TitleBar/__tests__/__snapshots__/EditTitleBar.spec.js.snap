// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTitleBar renders correctly when displayName is provided 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditTitleBar
    classes={
      Object {
        "description": "description",
        "input": "input",
        "itemSelector": "itemSelector",
        "section": "section",
        "title": "title",
        "titleDescription": "titledesc",
        "underline": "underline",
      }
    }
    description="The blue one"
    displayName="Regnbue Dash"
    name="Rainbow Dash"
    onChangeDescription={[Function]}
    onChangeTitle={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="titledesc"
        >
          <InputField
            className="title"
            filled={true}
            label="Dashboard title"
            name="Dashboard title input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="text"
            value="Rainbow Dash"
          />
          <TextField
            InputProps={
              Object {
                "classes": Object {
                  "root": "input",
                  "underline": "underline",
                },
              }
            }
            className="description"
            label="Dashboard description"
            multiline={true}
            name="Dashboard description input"
            onChange={[Function]}
            required={false}
            select={false}
            value="The blue one"
            variant="filled"
          />
        </div>,
        <div
          className="itemSelector"
        >
          <itemselector />
        </div>,
      ],
      "className": "section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <InputField
              className="title"
              filled={true}
              label="Dashboard title"
              name="Dashboard title input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="text"
              value="Rainbow Dash"
            />,
            <TextField
              InputProps={
                Object {
                  "classes": Object {
                    "root": "input",
                    "underline": "underline",
                  },
                }
              }
              className="description"
              label="Dashboard description"
              multiline={true}
              name="Dashboard description input"
              onChange={[Function]}
              required={false}
              select={false}
              value="The blue one"
              variant="filled"
            />,
          ],
          "className": "titledesc",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "title",
              "filled": true,
              "label": "Dashboard title",
              "name": "Dashboard title input",
              "onBlur": [Function],
              "onChange": [Function],
              "onFocus": [Function],
              "type": "text",
              "value": "Rainbow Dash",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "InputProps": Object {
                "classes": Object {
                  "root": "input",
                  "underline": "underline",
                },
              },
              "className": "description",
              "label": "Dashboard description",
              "multiline": true,
              "name": "Dashboard description input",
              "onChange": [Function],
              "required": false,
              "select": false,
              "value": "The blue one",
              "variant": "filled",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <itemselector />,
          "className": "itemSelector",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "itemselector",
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="titledesc"
          >
            <InputField
              className="title"
              filled={true}
              label="Dashboard title"
              name="Dashboard title input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="text"
              value="Rainbow Dash"
            />
            <TextField
              InputProps={
                Object {
                  "classes": Object {
                    "root": "input",
                    "underline": "underline",
                  },
                }
              }
              className="description"
              label="Dashboard description"
              multiline={true}
              name="Dashboard description input"
              onChange={[Function]}
              required={false}
              select={false}
              value="The blue one"
              variant="filled"
            />
          </div>,
          <div
            className="itemSelector"
          >
            <itemselector />
          </div>,
        ],
        "className": "section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <InputField
                className="title"
                filled={true}
                label="Dashboard title"
                name="Dashboard title input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="text"
                value="Rainbow Dash"
              />,
              <TextField
                InputProps={
                  Object {
                    "classes": Object {
                      "root": "input",
                      "underline": "underline",
                    },
                  }
                }
                className="description"
                label="Dashboard description"
                multiline={true}
                name="Dashboard description input"
                onChange={[Function]}
                required={false}
                select={false}
                value="The blue one"
                variant="filled"
              />,
            ],
            "className": "titledesc",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "title",
                "filled": true,
                "label": "Dashboard title",
                "name": "Dashboard title input",
                "onBlur": [Function],
                "onChange": [Function],
                "onFocus": [Function],
                "type": "text",
                "value": "Rainbow Dash",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "InputProps": Object {
                  "classes": Object {
                    "root": "input",
                    "underline": "underline",
                  },
                },
                "className": "description",
                "label": "Dashboard description",
                "multiline": true,
                "name": "Dashboard description input",
                "onChange": [Function],
                "required": false,
                "select": false,
                "value": "The blue one",
                "variant": "filled",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <itemselector />,
            "className": "itemSelector",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "itemselector",
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`EditTitleBar renders correctly when displayName not provided 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditTitleBar
    classes={
      Object {
        "description": "description",
        "input": "input",
        "itemSelector": "itemSelector",
        "section": "section",
        "title": "title",
        "titleDescription": "titledesc",
        "underline": "underline",
      }
    }
    description="The blue one"
    name="Rainbow Dash"
    onChangeDescription={[Function]}
    onChangeTitle={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="titledesc"
        >
          <InputField
            className="title"
            filled={true}
            label="Dashboard title"
            name="Dashboard title input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="text"
            value="Rainbow Dash"
          />
          <TextField
            InputProps={
              Object {
                "classes": Object {
                  "root": "input",
                  "underline": "underline",
                },
              }
            }
            className="description"
            label="Dashboard description"
            multiline={true}
            name="Dashboard description input"
            onChange={[Function]}
            required={false}
            select={false}
            value="The blue one"
            variant="filled"
          />
        </div>,
        <div
          className="itemSelector"
        >
          <itemselector />
        </div>,
      ],
      "className": "section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <InputField
              className="title"
              filled={true}
              label="Dashboard title"
              name="Dashboard title input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="text"
              value="Rainbow Dash"
            />,
            <TextField
              InputProps={
                Object {
                  "classes": Object {
                    "root": "input",
                    "underline": "underline",
                  },
                }
              }
              className="description"
              label="Dashboard description"
              multiline={true}
              name="Dashboard description input"
              onChange={[Function]}
              required={false}
              select={false}
              value="The blue one"
              variant="filled"
            />,
          ],
          "className": "titledesc",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "title",
              "filled": true,
              "label": "Dashboard title",
              "name": "Dashboard title input",
              "onBlur": [Function],
              "onChange": [Function],
              "onFocus": [Function],
              "type": "text",
              "value": "Rainbow Dash",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "InputProps": Object {
                "classes": Object {
                  "root": "input",
                  "underline": "underline",
                },
              },
              "className": "description",
              "label": "Dashboard description",
              "multiline": true,
              "name": "Dashboard description input",
              "onChange": [Function],
              "required": false,
              "select": false,
              "value": "The blue one",
              "variant": "filled",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <itemselector />,
          "className": "itemSelector",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "itemselector",
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="titledesc"
          >
            <InputField
              className="title"
              filled={true}
              label="Dashboard title"
              name="Dashboard title input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="text"
              value="Rainbow Dash"
            />
            <TextField
              InputProps={
                Object {
                  "classes": Object {
                    "root": "input",
                    "underline": "underline",
                  },
                }
              }
              className="description"
              label="Dashboard description"
              multiline={true}
              name="Dashboard description input"
              onChange={[Function]}
              required={false}
              select={false}
              value="The blue one"
              variant="filled"
            />
          </div>,
          <div
            className="itemSelector"
          >
            <itemselector />
          </div>,
        ],
        "className": "section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <InputField
                className="title"
                filled={true}
                label="Dashboard title"
                name="Dashboard title input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="text"
                value="Rainbow Dash"
              />,
              <TextField
                InputProps={
                  Object {
                    "classes": Object {
                      "root": "input",
                      "underline": "underline",
                    },
                  }
                }
                className="description"
                label="Dashboard description"
                multiline={true}
                name="Dashboard description input"
                onChange={[Function]}
                required={false}
                select={false}
                value="The blue one"
                variant="filled"
              />,
            ],
            "className": "titledesc",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "title",
                "filled": true,
                "label": "Dashboard title",
                "name": "Dashboard title input",
                "onBlur": [Function],
                "onChange": [Function],
                "onFocus": [Function],
                "type": "text",
                "value": "Rainbow Dash",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "InputProps": Object {
                  "classes": Object {
                    "root": "input",
                    "underline": "underline",
                  },
                },
                "className": "description",
                "label": "Dashboard description",
                "multiline": true,
                "name": "Dashboard description input",
                "onChange": [Function],
                "required": false,
                "select": false,
                "value": "The blue one",
                "variant": "filled",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <itemselector />,
            "className": "itemSelector",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "itemselector",
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
